lib/invoiced/util.rb

```diff
@@ -12,13 +12,26 @@ def uri_encode(params)
 
             def build_objects(_class, objects)
                 objects.map {
-                    |object| convert_to_object(_class, object)
+                    |object| convert_to_object(_class.client, _class, object)
                 }
             end
 
-            def convert_to_object(_class, values)
-                object = _class.class.new(_class.client, values[:id], values)
-                object.set_endpoint_base(_class.endpoint_base())
+            # arguments:
+            # client = client property of calling object
+            # _class = class of object to convert into
+            #   (has to be decoupled from `self` due to PaymentSource class)
+            # values = hash of values to populate object with
+            # base_link = if supplied, will set endpoint base to that of object called
+            def convert_to_object(client, _class, values, base_link=nil)
```

Comment:
Could the client be an optional argument at the end of the method here to prevent changing the signature for other callers of this function?

Also, I'm not sure I completely understand the comment about why client was required to be passed in. Can you explain that one further?

Reply:
Client has to be passed in all the time or the new object will not be associated with a connection. It can't be optional. In the past it was accessed by the calling object passing in `self` and the function grabbing the client off of it. This caused problems when creating a new object of a different class, although I just thought of something and will try a couple things to make this cleaner

Reply:
In those cases perhaps an optional client arg that falls back to the previous method when not specified would cause less changes

Reply:
Done